### PR TITLE
Quorum calculation to work with odd number of servers

### PR DESCRIPTION
**Pull request by TKffTK**
Afther pullrequest: https://github.com/minio/minio/pull/5428 for minio, we need to change also way to calculate quorum limits. Some discussion in that pullrequest.

Tested manually.

### CODE REVIEW

**Comment by harshavardhana** [comment]
You need to update the unit tests as well ?

**Comment by TKffTK** [comment]
My bad, will do.

File: dsync.go [hunk 1]
```diff
@@ -41,12 +42,10 @@ type Dsync struct {
 
 // New - initializes a new dsync object with input rpcClnts.
 func New(rpcClnts []NetLocker, rpcOwnNode int) (*Dsync, error) {
-	if len(rpcClnts) < 4 {
-		return nil, errors.New("Dsync is not designed for less than 4 nodes")
+	if len(rpcClnts) < 3 {
```
Comment (harshavardhana) [inline dsync.go]: We also decided to add support for 2 nodes where for write quorum we need 2 nodes available always.

File: dsync_test.go [hunk 2]
```diff
@@ -92,8 +92,8 @@ func TestMain(m *testing.M) {
 	}
 
 	nclnts = make([]NetLocker, 15)
-	if _, err := New(nclnts, 0); err == nil {
-		log.Fatalf("Should have failed")
+	if _, err := New(nclnts, 0); err != nil {
+		log.Fatalf("Should pass")
```
Comment (harshavardhana) [inline dsync_test.go]: you should add more tests to check for odd number of nodes.. validate the dquorum and dquorumReads values as well.

File: dsync.go (hunk 1 above)
Comment (harshavardhana) [reply dsync.go]: Additionally we are also moving to maximum of 32 nodes not 16 we need to change that as well and associated tests.

**Comment by harshavardhana** [comment]
sent a PR to add tests @TKffTK https://github.com/remodoy/dsync/pull/1

**Comment by harshavardhana** [comment]
Changes have been merged.. closing this PR.

**Pull request closed without merging**